x/gov/keeper/vote.go

```diff
@@ -68,11 +68,23 @@ func (keeper Keeper) GetVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.A
 	}
 
 	keeper.cdc.MustUnmarshal(bz, &vote)
+
+	// Graceful fallback of deprecated `Option` field, in case there's only 1
+	// VoteOption.
+	if len(vote.Options) == 1 && vote.Options[0].Weight.Equal(sdk.MustNewDecFromStr("1.0")) {
+		vote.Option = vote.Options[0].Option //nolint staticcheck // Deprecated field
+	}
+
 	return vote, true
 }
 
 // SetVote sets a Vote to the gov store
 func (keeper Keeper) SetVote(ctx sdk.Context, vote types.Vote) {
+	// vote.Option is a deprecated field, we don't set it in state
+	if vote.Option != types.OptionEmpty { //nolint staticcheck // Deprecated field
+		panic(fmt.Errorf("expected empty vote.Option, got %s", vote.Option)) //nolint staticcheck // Deprecated field
+	}
```

Comment:
@robert-zaremba 

> In Keeper.AddVote, shall we check that Vote.Option and Vote.Options are exclusive (on can't set both parameters)?

AddVote takes `options` as argument, I don't think there's anything to change.

I added this piece of code, so that in state, we always only store the weighted option, does that seem okay to you?